photon-core/src/main/java/org/photonvision/common/configuration/NeuralNetworkModelManager.java

```diff
@@ -18,87 +18,250 @@
 package org.photonvision.common.configuration;
 
 import java.io.File;
-import java.io.FileOutputStream;
 import java.io.IOException;
+import java.net.URL;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
 import java.util.List;
 import org.photonvision.common.logging.LogGroup;
 import org.photonvision.common.logging.Logger;
 import org.photonvision.rknn.RknnJNI;
 
+/**
+ * Manages the loading of neural network models.
+ *
+ * <p>Models are loaded from the filesystem at the <code>modelsFolder</code> location. PhotonVision
+ * also supports shipping pre-trained models as resources in the JAR. If the model has already been
+ * extracted to the filesystem, it will not be extracted again.
+ *
+ * <p>Each model must have a corresponding <code>labels</code> file. The labels file format is
+ * simply a list of string names per label, one label per line. The labels file must have the same
+ * name as the model file, but with the suffix <code>-labels.txt</code> instead of <code>.rknn
+ * </code>.
+ *
+ * <p>Note: PhotonVision currently only supports YOLOv5 and YOLOv8 models in the <code>.rknn</code>
+ * format.
+ */
 public class NeuralNetworkModelManager {
+    /** Singleton instance of the NeuralNetworkModelManager */
     private static NeuralNetworkModelManager INSTANCE;
-    private static final Logger logger = new Logger(NeuralNetworkModelManager.class, LogGroup.Config);
 
-    private final String MODEL_NAME = "note-640-640-yolov5s.rknn";
-    private final RknnJNI.ModelVersion modelVersion = RknnJNI.ModelVersion.YOLO_V5;
-    private File defaultModelFile;
-    private List<String> labels;
+    /**
+     * Private constructor to prevent instantiation
+     *
+     * @return The NeuralNetworkModelManager instance
+     */
+    private NeuralNetworkModelManager() {}
 
+    /**
+     * Returns the singleton instance of the NeuralNetworkModelManager
+     *
+     * @return The singleton instance
+     */
     public static NeuralNetworkModelManager getInstance() {
         if (INSTANCE == null) {
             INSTANCE = new NeuralNetworkModelManager();
         }
         return INSTANCE;
     }
 
+    /** Logger for the NeuralNetworkModelManager */
+    private static final Logger logger = new Logger(NeuralNetworkModelManager.class, LogGroup.Config);
+
+    /**
+     * Determines the model version based on the model's filename.
+     *
+     * <p>"yolov5" -> "YOLO_V5"
+     *
+     * <p>"yolov8" -> "YOLO_V8"
+     *
+     * @param modelName The model's filename
+     * @return The model version
+     */
+    private static RknnJNI.ModelVersion getModelVersion(String modelName)
```

Comment:
This seems like an acceptable hack for now. I'm not sure of a better way to maintain metadata about known uploaded models without having to deal with syncing between a database and the filesystem.

Reply:
I think long term `RknnJNI.ModelVersion` here needs to replaced with a product type `(Version, Format)`, ie (`YOLO_V8`, `RKNN`).

As for the hack - I think encoding this kind of metadata in the filename is going to be for the best for now. It might be worth reconsidering when/if we get around to model upload UI.